extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -50,11 +50,37 @@ export class CreateProjectFromDatabaseDialog {
 
 		this.dialog.cancelButton.label = constants.cancelButtonText;
 
+		let connected = false;
+		if (this.profile) {
+			const connections = await azdataType.connection.getConnections(true);
+			connected = !!connections.find(c => c.connectionId === this.profile!.id);
+
+			if (!connected) {
+				// if the connection clicked on isn't currently connected, try to connect
+				const result = await azdataType.connection.connect(this.profile, true, false);
+				connected = result.connected;
+
+				if (!result.connected) {
+					// if can't connect automatically, open connection dialog with the info from the profile
+					const connection = await azdataType.connection.openConnectionDialog(undefined, this.profile);
+					connected = !!connection;
+
+					// update these fields if connection was successful, to ensure they match the connection made
+					if (connected) {
+						this.profile.id = connection.connectionId;
+						this.profile.databaseName = connection.options['databaseName'];
+						this.profile.serverName = connection.options['server'];
+						this.profile.userName = connection.options['user'];
```

Comment:
this is the exhaustive list - any chance we could miss copying something from connection object?

Reply:
these are only used for displaying the connection in the dialog. The connection id is what actually gets used for creating the project